### ISSUE TITLE
Currency reading in excel.

### ISSUE THREAD

**Issue by fernando-jose-silva** [nvaccess/nvda#7263]
The nvda when reading a value in excel when this number is formatted as currency utters the dot character.
To reproduce:
For the tests I used windows, nvda, synthesizer and excel in Portuguese Brazil.
Create a worksheet in excel and enter the value:
2468,87
Now format this cell as currency.
When reading the cell with the arrows see that the nvda reads:
R$2.468,87
When the correct one would be to pronounce:
R$2 mil 468,87.
Thank you.

**Comment by dkager** [comment]
Is this specific to the synthesizer in use? For me eSpeak doesn't read it differently if I copy and paste to Notepad.

**Comment by fernando-jose-silva** [comment]
The problem here is that in Portuguese it would be correct for the number below to be read as
R$ 2.357,72
real 2 mil 357 vírgula (comma) 72.

What happens is that the number is read as


r$ 2 ponto (.) 357 vírgula (comma) 72


This by pasting into the notepad and using the espeak

**Comment by bhavyashah** [comment]
CC @dkager  in case you have any follow-up comment to @fernando-jose-silva 's https://github.com/nvaccess/nvda/issues/7263#issuecomment-313228550. I am unable to ascertain from comments whether this issue is on the part of the speech synthesizer or the fault of the screen reader. Any additional inputs? @Qchristensen

**Comment by Qchristensen** [comment]
Logically, the value R$ 2.357,72 might be understood (by a human) to mean
"real 2 mil 357 vírgula (comma) 72", however what is written is R$ 2.357,72 so that is what is read.

Just as in US currency $2.95 means "Two dollars and ninety-five cents" to a human, what will be read out is "dollar 2 point nine-five".

Theoretically, you could override this with dictionary rules in NVDA using regex. I'm not sure I'd call it a bug as NVDA in general doesn't try to interpret numbers too much, rather than just send them to the synthesizer as is because there are so many different ways they might be interpreted.

**Comment by fernando-jose-silva** [comment]
Changing nvda's speech to read coins in a way to the user's taste by changing the nvda dictionary for beginners is not very easy.
My suggestion would be so that nvda could offer the user a way to read more friendly coins.

Enviado do meu iPhone

> Em 4 de ago de 2017, às 05:51, Quentin Christensen <notifications@github.com> escreveu:
> 
> Logically, the value R$ 2.357,72 might be understood (by a human) to mean
> "real 2 mil 357 vírgula (comma) 72", however what is written is R$ 2.357,72 so that is what is read.
> 
> Just as in US currency $2.95 means "Two dollars and ninety-five cents" to a human, what will be read out is "dollar 2 point nine-five".
> 
> Theoretically, you could override this with dictionary rules in NVDA using regex. I'm not sure I'd call it a bug as NVDA in general doesn't try to interpret numbers too much, rather than just send them to the synthesizer as is because there are so many different ways they might be interpreted.
> 
> —

**Comment by Adriani90** [comment]
@fernando-jose-silva are you still having this issue in NVDA 2019.1.1? If yes, which synthesizer are you using? Is the same issue with other synthesizers reproducible? I had similar problems with a synthesizer which added entries to my speech dictionary without warning me before. I discovered that after hours of investigating because reportings in MS Excel were strongly broken (i.e. NVDA reported in some cases cell coordinate before content etc.). So if you are still having this issue, please try to clean you speech dictionary and see if it still happening.

**Comment by Adriani90** [comment]
For me, I cannot reproduce this issue in german, romanian and english language with different synthesizers.

**Comment by fernando-jose-silva** [comment]
Perfect
@Adriani90 
Solved deleting multiple entries of dictionaries, the vast majority not created by me.
Congratulations on your work, very valuable for everyone.

**Comment by Adriani90** [comment]
Thank you very much,
closing as works for me.
